### PR TITLE
Bug-295 - tests for gtest/gmock env

### PR DESCRIPTION
**Pull request by spamaa**
These are tests, that checks does gtest/gmock basic features works. When we will use a new kind of gmock/gtest feature, we should add a test that verifies if it works at all enviroments.

### CODE REVIEW

File: tests/dummy/Interfaces/DummyInterface.hpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2010-2022 The Open-Transactions developers
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#if !defined (__TEST_DUMMY_INTERFACES_DUMMYINTERFACE_HPP_INCLUDED__)
```
Comment (justusranvier) [inline tests/dummy/Interfaces/DummyInterface.hpp:6]: Use #pragma once instead of header guards

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -405,6 +405,9 @@ if(OT_INSTALL_LIBRARY_DEPENDENCIES)
   endif()
 
   if(OPENTXS_BUILD_TESTS)
+    # TEST_BEGIN
+
+    # TEST_END
```
Comment (justusranvier) [inline src/CMakeLists.txt:410]: What are these comments placeholders for?

**Review by justusranvier — CHANGES_REQUESTED** [review]
See [this comment](https://github.com/Open-Transactions/opentxs-cmake/pull/1#issuecomment-1047025184) for alternatives to universally disabling Weffc++ in all our projects.

The best approach is to use the #pragma GCC technique since modifying the compiler flags for a specific file makes unity builds less effective but if that is too unwieldy turning off Weffc++ on the specific test files that use gmock is also acceptable.

File: tests/dummy/Interfaces/DummyInterface.hpp (hunk 1 above)
Comment (spamaa) [reply tests/dummy/Interfaces/DummyInterface.hpp:6]: #pragma once  works only at compilers that support it, and usually ignored by compilers that do not support it. If it is not supported, compilers usually will not give even warning. At the other hand, define is more error prone. You can mismatch defines, but it almost always produces easy to detect compilation errors. Hence I prefer the define guards. Code is more robust and cost almost nothing. I will change it to keep consistent with remaining code.

File: src/CMakeLists.txt (hunk 2 above)
Comment (spamaa) [reply src/CMakeLists.txt:410]: Sorry, I was using some debug code and it was just for clear identification what is a debug code and what isn't. I have forgot to delete it.

Comment (spamaa) [reply src/CMakeLists.txt]: done

**Pull request merged**